mongo/src/main/scala/io/sphere/mongo/format/DefaultMongoFormats.scala

```diff
@@ -140,4 +141,86 @@ trait DefaultMongoFormats {
     }
   }
 
+  implicit val currencyFormat: MongoFormat[Currency] = new MongoFormat[Currency] {
+    val failMsg = "ISO 4217 code JSON String expected"
+
+    override def toMongoValue(c: Currency): Any = c.getCurrencyCode
+    override def fromMongoValue(any: Any): Currency = any match {
+      case s: String =>
+        try {
+          Currency.getInstance(s)
+        } catch {
+          case _: IllegalArgumentException => throw new Exception(failMsg)
```

Comment:
I generally like to see the input that caused a validation failure in the Exception. Do you think its worth it here to include `s`?

Reply:
I also like that.
I used the same strategy as in json.
Maybe I can add it to both.

Reply:
https://github.com/sphereio/sphere-scala-libs/pull/79/commits/f88e868483c68dceffc729b748bd2ec27281ec50